html/Axit/js/validation/helper-functions.js

```diff
@@ -1,113 +1,180 @@
+    return formatResult;
   }
-  localError.innerHTML = '';
-  return true;
+  return '';
 }
 
-function isInputNotEmpty(input, errorInput, errorMessage) {
-  const localErrorInput = errorInput;
+function isInputNotEmpty(input) {
   if (input.value.trim().length === 0) {
-    localErrorInput.textContent = errorMessage;
-    return false;
+    return 'This field is required';
   }
-  return true;
+  return '';
 }
 
-function isNameCorrect(input, errorInput, errorMessage) {
+function isNameCorrect(input) {
   const nameRegex = /^[a-zA-Z]+$/g;
-  const localErrorInput = errorInput;
   if (!input.value.trim().match(nameRegex)) {
-    localErrorInput.textContent = errorMessage;
-    return false;
+    return 'Only latin letters';
   }
-  return true;
+  return '';
 }
 
-function isMessageCorrect(input, errorInput, errorMessage) {
-  const localErrorInput = errorInput;
+function isMessageCorrect(input) {
   if (input.value.trim().length < 1 || input.value.trim().length > 255) {
-    localErrorInput.textContent = errorMessage;
-    return false;
+    return 'Message should have min 1m max 250 characters';
   }
-  return true;
+  return '';
 }
 
-function isEmailFormatCorrect(input, errorInput, errorMessage) {
-  const localErrorInput = errorInput;
+function isEmailFormatCorrect(input) {
   const emailRegex = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
   if (!input.value.trim().match(emailRegex) || input.value.trim().length < 6) {
-    localErrorInput.textContent = errorMessage;
-    return false;
+    return 'Email is invalid';
   }
-  return true;
+  return '';
 }
 
-function isPasswordCorrect(input, errorInput, errorMessage) {
-  const localErrorInput = errorInput;
+function isPasswordCorrect(input) {
   const passwordRegex = /^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[#?!@$%^&*-]).{6,}$/;
   if (!input.value.trim().match(passwordRegex)) {
-    localErrorInput.textContent = errorMessage;
+    return 'Min 6 characters, required: 1 capital letter, 1 lower-case letter, 1 digit, 1 special character';
+  }
+  return '';
+}
+
+export function findErrorElement(errorMessages, errorName) {
+  return Array.from(errorMessages).find(function(element) {
+    return element.dataset.errorName === errorName;
+  });
+}
+
+function setError(error, message) {
+  // eslint-disable-next-line no-param-reassign
+  error.innerHTML = message;
+}
+
+function showInputStatus(validationResult, errorMessages, errorName) {
+  const error = findErrorElement(errorMessages, errorName);
+  setError(error, validationResult);
+  if (validationResult.length !== 0) {
     return false;
+  } else {
+    return true;
+  }
+}
+
+export function validateRegistrationForm({
+  bannerForm,
+  errorMessages,
+  nameInput,
+  emailInput,
+  passwordInput,
+}) {
+  let result = [];
+
+  // Name validation
+  const nameValidationResult = checkName(nameInput);
```

Comment:
it would be better if these functions return true/false and after by field name you could chouse correct errorMessage to show for example form object with messages